### PR TITLE
Add test for sprint (fixes: #45066)

### PR DESCRIPTION
**Pull request by vmpyr**
Added tests for sprint in `test/strings/io.jl` since the function is defined in `base/strings/io.jl`. Refer: #45277 & #45066

### CODE REVIEW

**Comment by vmpyr** [comment]
@vtjnash, can you please review this?

File: test/strings/io.jl [hunk 1]
```diff
@@ -219,6 +219,10 @@ end
         """
 end
 
+@testset "sprint honoring IOcontext" begin
+    @test sprint(show, Base.Dict[], context=(:compact=>false, :module=>nothing))[1:4] === "Base"
```
Comment (vtjnash) [inline test/strings/io.jl:223]:
```suggestion
@testset "sprint honoring IOContext" begin
    @test startswith(sprint(show, Base.Dict[], context=(:compact=>false, :module=>nothing)), "Base.Dict")
```

**Review by vtjnash — APPROVED** [review]
Thanks!

n.b. the issue was fixed by #46716, but was missing this test to close the issue @KristofferC

File: test/strings/io.jl (hunk 1 above)
Comment (vmpyr) [reply test/strings/io.jl]: I have made the suggested changes!

**Pull request merged**